salt/_modules/azurearm_authorization.py

```diff
@@ -0,0 +1,555 @@
+    * `azure-common <https://pypi.python.org/pypi/azure-common>`_ >= 1.1.8
+    * `azure-mgmt <https://pypi.python.org/pypi/azure-mgmt>`_ >= 1.0.0
+    * `azure-mgmt-compute <https://pypi.python.org/pypi/azure-mgmt-compute>`_ >= 1.0.0
+    * `azure-mgmt-network <https://pypi.python.org/pypi/azure-mgmt-network>`_ >= 1.7.1
+    * `azure-mgmt-resource <https://pypi.python.org/pypi/azure-mgmt-resource>`_ >= 1.1.0
+    * `azure-mgmt-storage <https://pypi.python.org/pypi/azure-mgmt-storage>`_ >= 1.0.0
+    * `azure-mgmt-web <https://pypi.python.org/pypi/azure-mgmt-web>`_ >= 0.32.0
+    * `azure-storage <https://pypi.python.org/pypi/azure-storage>`_ >= 0.34.3
+    * `msrestazure <https://pypi.python.org/pypi/msrestazure>`_ >= 0.4.21
+:platform: linux
+
+:configuration: This module requires Azure Resource Manager credentials to be passed as keyword arguments
+to every function in order to work properly.
+
+    Required provider parameters:
+
+    if using username and password:
+      * ``subscription_id``
+      * ``username``
+      * ``password``
+
+    if using a service principal:
+      * ``subscription_id``
+      * ``tenant``
+      * ``client_id``
+      * ``secret``
+
+    Optional provider parameters:
+
+    **cloud_environment**: Used to point the cloud driver to different API endpoints, such as Azure GovCloud.
+    Possible values:
+      * ``AZURE_PUBLIC_CLOUD`` (default)
+      * ``AZURE_CHINA_CLOUD``
+      * ``AZURE_US_GOV_CLOUD``
+      * ``AZURE_GERMAN_CLOUD``
+
+'''
+
+# Python libs
+from __future__ import absolute_import
+import logging
+
+# Azure libs
+HAS_LIBS = False
+try:
+    import azure.mgmt.authorization.models  # pylint: disable=unused-import
+    from msrest.exceptions import SerializationError
+    from msrestazure.azure_exceptions import CloudError
+    HAS_LIBS = True
+except ImportError:
+    pass
+
+__virtualname__ = 'azurearm_authorization'
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    if not HAS_LIBS:
+        return (
+            False,
+            'The following dependencies are required to use the AzureARM modules: '
+            'Microsoft Azure SDK for Python >= 2.0rc6, '
+            'MS REST Azure (msrestazure) >= 0.4'
+        )
+
+    return __virtualname__
+
+
+def provider_operations_metadata_get(resource_provider_namespace, **kwargs):
+    '''
+    .. versionadded:: Sodium
+
+    Gets provider operations metadata for the specified resource provider.
+
+    :param resource_provider_namespace: The namespace of the resource provider.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-call azurearm_authorization.provider_operations_metadata_get testnamespace
+
+    '''
+    result = {}
+    authconn = __utils__['azurearm.get_client']('authorization', **kwargs)
+    try:
+        data = authconn.provider_operations_metadata.get(
+            resource_provider_namespace=resource_provider_namespace,
+            api_version='2015-07-01',
```

Comment:
I did this originally because the one module you had completed already had api_version like this. However, I decided to keep doing it like that because if you change the API version for some of these functions the other parameter names change, which can be challenging. Do you want me to make the variable api_version still and then put conditionals in place for the different versions?

Reply:
nothing that fancy.
`def provider_operations_metadata_get(resource_provider_namespace, api_version='2015-07-01', **kwargs):`
then just use that variable on line 100
people can override at their own risk